src/shared/ml-thirdparty-client/index.ts

```diff
@@ -157,4 +158,123 @@ export class Client {
 
     // TODO: Implement communication with Mojaloop.
   }
+
+  /**
+   * Gets a list of PISP/DFSP participants
+   */
+  public async getParticipants(): Promise<void> {
+    if (this.simulator) {
+      // If the client is configured with a simulator, then it will not
+      // communicate with Mojaloop directly. Instead, it will only generate
+      // a random response that is injected to the internal routes.
+      return this.simulator.getParticipants()
+    }
+    // TODO: Add once implemented in sdk-standard components
+  }
+
+  /**
+   * Performs a request for a new consent
+   *
+   * @param requestBody         an consent request object as defined by the Mojaloop API.
+   * @param destParticipantId   ID of destination - to be used when sending request
+   */
+  public async postConsentRequests(
+    requestBody: SDKStandardComponents.PostConsentRequestsRequest,
+    destParticipantId: string
+  ): Promise<void> {
+    if (this.simulator) {
+      // If the client is configured with a simulator, then it will not
+      // communicate with Mojaloop directly. Instead, it will only generate
+      // a random response that is injected to the internal routes.
+      return this.simulator.postConsentRequests(requestBody)
+    }
+    await this.thirdpartyRequests.postConsentRequests(
+      requestBody,
+      destParticipantId
+    )
```

Comment:
I think generally this is something best handled by dependency injection. E.g., you provide a library (such as `this.thirdpartyRequests`) and then do the same actions on it...

So ultimately, this code would be something like

```typescript
return this._requestClass.postConsentRequests(requestBody, destParticipantId);
```

And the way you'd "turn on the simulator" would be by setting `_requestClass` to what you currently have in `this.simulator`...

Reply:
so essentially just separating the two classes completely? in which case we're just using simulator for testing and it wouldn't be part client library at all, right?